lib/payment/merchant_stripe.rb

```diff
@@ -0,0 +1,182 @@
+module ShorePayment
+  #
+  class StripeHash
+    def initialize(attrs = {})
+      update_attributes(attrs)
+    end
+
+    # Update object with this method. We have to take care updating 'nested'
+    #   objects
+    def update_attributes(attrs = {})
+      attrs.each_pair do |attr, value|
+        send(:"#{attr}=", value) if respond_to?(:"#{attr}=")
+      end if attrs
+    end
+  end
+
+  # Conversion between day of birth's Hash and Date representation
+  module DobConvertible
+    def dob_date
+      if @dob && dob_present?
+        Date.new(@dob.year.to_i, @dob.month.to_i, @dob.day.to_i)
+      end
+    end
+
+    def dob_date=(new_date)
+      date = new_date.to_date
+      @dob = DateOfBirth.new(
+        year: date.year,
+        month: date.month,
+        day: date.day
+      ) if date
+    end
+
+    private
+
+    def dob_present?
+      @dob.year.present? && @dob.month.present? && @dob.day.present?
+    end
+  end
+
+  # Representation of a {DOB} object in the Organization response
+  class DateOfBirth < StripeHash
+    attr_accessor :day, :month, :year
+  end
+
+  # Representation of an {Address} object in the Organization response
+  class Address < StripeHash
+    attr_accessor :city, :country, :line1, :line2, :postal_code, :state
+  end
+
+  # Representation of an {AdditionalOwner} object in the Organization response
+  class AdditionalOwner < StripeHash
+    include DobConvertible
+
+    attr_accessor :first_name, :last_name, :dob
+
+    def dob=(attrs)
+      @dob = DateOfBirth.new(attrs)
+    end
+  end
+
+  # Representation of a {LegalEntity} object in the Organization response
+  class LegalEntity < StripeHash
+    include DobConvertible
+
+    attr_accessor :additional_owners, :address, :dob, :first_name, :last_name,
+                  :type
+
+    def update_attributes(attrs = {})
+      super
+      # We have to set additional owners to an empty string to delete
+      #   previously added additional owners.
+      @additional_owners = '' if @type == 'individual' || number_of_owners == 1
+    end
+
+    def address=(attrs)
+      @address = Address.new(attrs)
+    end
+
+    def additional_owners=(attrs)
+      # We always build a new array of additional owners, because there is no
+      #   such thing as editing a single additional owner.
+      # Setting additional_owners to an empty string means that stripe API
+      #   should delete all previously added additional owners
+      @additional_owners = if !attrs.present?
+                             attrs
+                           else
+                             attrs = attrs.values if attrs.respond_to?(:values)
```

Comment:
I'm leaving this here now, because it's required for to handle the case when you change the type to 'individual' but I'll try to find out something better.